shotover-proxy/src/transforms/cassandra/sink_cluster/node_pool.rs

```diff
@@ -0,0 +1,95 @@
+use super::routing_key::calculate;
+use super::token_map::TokenMap;
+use crate::transforms::cassandra::sink_cluster::node::CassandraNode;
+use cassandra_protocol::frame::message_execute::BodyReqExecuteOwned;
+use cassandra_protocol::frame::message_result::PreparedMetadata;
+use cassandra_protocol::frame::Version;
+use cassandra_protocol::token::Murmur3Token;
+use rand::prelude::*;
+use std::collections::HashMap;
+
+#[derive(Debug)]
+pub struct NodePool {
+    prepared_metadata: HashMap<Vec<u8>, PreparedMetadata>,
+    token_map: TokenMap,
+    pub nodes: Vec<CassandraNode>,
+
+    rng: SmallRng,
```

Comment:
we have two sources of rng now.
we should either pass the rng into NodePool methods when we need it here or use node_pool.rng when we need it in the transform.